eth-contracts/contracts/staking/Staking.sol

```diff
@@ -161,20 +166,7 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IsContract {
      * @param _data Used in Unstaked event, to add signalling information in more complex staking applications
      */
     function unstake(uint256 _amount, bytes calldata _data) external isInitialized {
-        // unstaking 0 tokens is not allowed
-        require(_amount > 0, ERROR_AMOUNT_ZERO);
-
-        // Adjust amount by internal stake multiplier
-        uint internalStakeAmount = _amount.div(stakeMultiplier.getLatestValue());
-
-        // checkpoint updated staking balance
-        _modifyStakeBalance(msg.sender, internalStakeAmount, false);
-
-        // checkpoint total supply
-        _modifyTotalStaked(_amount, false);
-
-        // transfer tokens
-        stakingToken.safeTransfer(msg.sender, _amount);
+        _unstakeInternal(_amount);
```

Comment:
the reason i had to move this is unstake is an external function and also has bytes calldata which i don't need for the withdrawFunds call. So just moved this logic into an internal function called by both

Reply:
makes sense to me